### PR TITLE
Boom update

### PR DESCRIPTION
**Pull request by paciops**
- Migrated from [`boom`](https://www.npmjs.com/package/boom) to [`@hapi/boom`](https://www.npmjs.com/package/@hapi/boom)
- Introduced tests for `app()` and `errorMiddleware`
- Introduced CI with lint and tests
- Updated `publish` with node 18

### CODE REVIEW

File: .github/workflows/lint-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.version }}
+
+      - run: npm ci
+      - run: npm run lint
+      - run: npm t
```
Comment (ksnll) [inline .github/workflows/lint-and-test.yml:21]: missing a newline

File: lib/errorMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+function dispatch(err, req, res) {
+	res.status(err.output.statusCode);
+	if (req.accepts("json"))
+		res
+			.set("Content-Type", "application/json")
+			.send(`${JSON.stringify(err.output.payload)}\n`);
+	else if (req.accepts(["application/xml", "text/xml"]))
+		res.set("Content-Type", "text/xml").send(`${xml(err.output.payload)}\n`);
+	else
+		res.send(
```
Comment (ksnll) [inline lib/errorMiddleware.js:15]: formatting looks really off. can we maybe introducing biome?

File: lib/errorMiddleware.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+			format(
+				"%s %s\n%s\n",
+				err.output.payload.statusCode,
+				err.output.payload.message,
+				err.output.payload.stack || "",
```
Comment (ksnll) [inline lib/errorMiddleware.js:20]: we now send the stack trace no matter what, in the previous version, we only sent it based on a configuration

File: test/boom.test.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+const { strictEqual, deepStrictEqual } = require("node:assert/strict");
+const newBoom = require("@hapi/boom");
+
+/**
+ * This test is here to prove that the old Boom.wrap method
+ * is equal to the new Boom class.
+ * This test exit with 0 if the boom library is not installed.
+ * It should not be installed since it is deprecated https://www.npmjs.com/package/boom
+ */
+describe("Booom testing", () => {
```
Comment (ksnll) [inline test/boom.test.js:11]: I'm unsure what this test is actually proving? given by default, we don't import boom

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -10,10 +10,10 @@ jobs:
       contents: read
       packages: write
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
         with:
-          node-version: '16.x'
+          node-version: '18.x'
```
Comment (fmaule) [inline .github/workflows/publish.yml:16]: why are we using node 18.x instead of 20.x ?

**Review by ksnll — APPROVED** [review]
🍏

File: .github/workflows/publish.yml (hunk 5 above)
Comment (paciops) [reply .github/workflows/publish.yml:16]: i am not sure if every app that will use this module is using node 20

File: lib/errorMiddleware.js (hunk 2 above)
Comment (paciops) [reply lib/errorMiddleware.js:15]: yes I would like to introduce biome in a next PR

File: test/boom.test.js (hunk 4 above)
Comment (paciops) [reply test/boom.test.js:11]: It is proving that the new Boom class works exactly as the old one

**Pull request merged**